lib/timer.py

```diff
@@ -139,7 +140,13 @@ def run_jobs(cfg: UAConfig, current_time: datetime):
     Persist jobs-status with calculated next_run values to aid in timer
     state introspection for jobs which have not yet run.
     """
-    jobs_status_obj = timer_jobs_state_file.read()
+    jobs_status_obj = None
+    # If the file format is wrong we remove it, and after the jobs are
+    # executed it will be recreated with the proper data.
+    try:
+        jobs_status_obj = timer_jobs_state_file.read()
+    except InvalidFileFormatError:
+        timer_jobs_state_file.delete()
```

Comment:
I can see scenarios where even deletion fails.
We can't do much in those cases ...
But - while we are at trying to change crashes into proper errors - should we change it to also gracefully die if it fails to remove it?

I see that down in the stack `ensure_file_absent` will prevent `FileNotFoundError` from ever happening, but `OSError` and `PermissionError` might still occur.

Reply:
Good catch - I will update this to fail gracefully in that case.
But then, what should we do? Always run the jobs when the timer triggers, because we don't know when they ran last time? or don't run the jobs at all?